src/core/thread/key_manager.hpp

```diff
@@ -152,6 +153,48 @@ class NetworkKey : public otNetworkKey, public Equatable<NetworkKey>
      */
     Error GenerateRandom(void) { return Random::Crypto::FillBuffer(m8, sizeof(m8)); }
 #endif
+
+} OT_TOOL_PACKED_END;
+
+/**
+ * Provides a representation for Network Key reference.
+ *
+ */
+typedef otNetworkKeyRef NetworkKeyRef;
+
+/**
+ * This class represents a Thread Network Key material.
+ *
+ */
+OT_TOOL_PACKED_BEGIN
```

Comment:
Thinking mroe on this, can we consider simplifying and not even having a `NetworkKeyInfo` struct.
- In `KeyManager` class, if `OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE` then we have a priavte `NetworkKeyRef mNetworkKeyRef`, otherwise we save it as `NetworkKey mNetworkKey`.
- Then let the different flavors of getters and setters to handle different cases (i.e. `GetNetworkKey()/SetNetworkKey() for literal key and `GetNetworkKeyRef()/SetNetworkKeyRef()` when PSA is enabled and want to get the key ref).

Reply:
@abtink sure, I will revert the changes and go back to original implementation of having KeyRefs under conditional compilation.

Reply:
I will also remove the CopyKey and bring back the Getters which abstract the PSA usage and return the Key irrespective of what is being used internally.